iceoryx_utils/include/iceoryx_utils/internal/units/duration.inl

```diff
@@ -21,47 +21,102 @@ namespace iox
 namespace units
 {
 template <typename T>
-constexpr Duration Duration::nanoseconds(const T ns)
+constexpr Duration Duration::nanoseconds(const T value)
 {
-    return operator"" _ns(ns);
+    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
+
+    if (value < 0)
+    {
+        return Duration{0U, 0U};
+    }
+    return operator"" _ns(value);
 }
 template <typename T>
-constexpr Duration Duration::microseconds(const T us)
+constexpr Duration Duration::microseconds(const T value)
 {
-    return operator"" _us(us);
+    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
+
+    if (value < 0)
+    {
+        return Duration{0U, 0U};
+    }
+    return operator"" _us(value);
 }
 template <typename T>
-constexpr Duration Duration::milliseconds(const T ms)
+constexpr Duration Duration::milliseconds(const T value)
 {
-    return operator"" _ms(ms);
+    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
+
+    if (value < 0)
+    {
+        return Duration{0U, 0U};
+    }
+    return operator"" _ms(value);
 }
 template <typename T>
-constexpr Duration Duration::seconds(const T seconds)
+constexpr Duration Duration::seconds(const T value)
 {
-    return operator"" _s(seconds);
+    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
+
+    if (value < 0)
+    {
+        return Duration{0U, 0U};
+    }
+    return operator"" _s(value);
 }
 template <typename T>
-constexpr Duration Duration::minutes(const T min)
+constexpr Duration Duration::minutes(const T value)
 {
-    return operator"" _m(min);
+    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
+
+    if (value < 0)
+    {
+        return Duration{0U, 0U};
+    }
+    return operator"" _m(value);
 }
 template <typename T>
-constexpr Duration Duration::hours(const T hours)
+constexpr Duration Duration::hours(const T value)
 {
-    return operator"" _h(hours);
+    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
+
+    if (value < 0)
+    {
+        return Duration{0U, 0U};
+    }
+    return operator"" _h(value);
 }
 template <typename T>
-constexpr Duration Duration::days(const T days)
+constexpr Duration Duration::days(const T value)
+{
+    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
+
+    if (value < 0)
+    {
+        return Duration{0U, 0U};
+    }
+    return operator"" _d(value);
+}
+
+
+inline constexpr Duration::Duration(const uint64_t seconds, const uint32_t nanoseconds)
+    : m_seconds(seconds)
+    , m_nanoseconds(nanoseconds)
 {
-    return operator"" _d(days);
+    if (nanoseconds >= 1000000000U)
```

Comment:
`1000000000U` is used quite often as conversion factor in the code, maybe it make sense to make a constexpr of it?
Proposal: `CONV_NANOSECONDS`